documentation/Architecture/Overview.md

```diff
@@ -2,80 +2,105 @@
-Below is a brief overview of the structure of the SDK.
+<img src=".attachments/overview.svg" width="800">
 
 
-![](.attachments/ArchitectureOverview.png)
+The interfaces depicted in this diagram are
 
+1) The SDK takes as input *data sources* (such as file paths) from the *SDK driver*. This input is typically generated from the user, such as when the user asks 
+the driver to open a file.
+2) The SDK passes the data sources to all *SDK plugins* which can support them. Plugins are 
+loaded dynamically at runtime by the SDK driver before the data sources are given to the SDK. In this example, 2 out of the 3 loaded plugins advertise that 
+they can handle the loaded data source.
+3) Plugins can optionally exchange data through *data cookers*, even if the two plugins do not have access to each others' source code. In this 
+example, plugin B queries for and receives data from plugin A. Note that this data can be programmatically 
+accessed by anyone - not just other plugins. See the Plugins section for more information.
+4) Each plugin given the data sources returns *tables* to the SDK. Note that a plugin which advertises support for a data source may return no tables, but that 
+is unlikely to happen in practice.
+5) The SDK outputs all tables created by plugins to the SDK driver.
+6) The SDK driver does whatever it wishes with the tables. In WPA, the tables are rendered as interactable and manipulable graphs.
 
-# Step-by-step
-1) Select Data Source in Driver
-2) Driver passes Data Source to SDK
-3) SDK passes Data Source to the Custom Data Source
-4) Custom Data Source validates Data Source and sends plugin (compiled binaries) to SDK
-5) Plugins create default/custom tables which the SDK dynamically sends to the Driver
-6) Driver displays visualized data for interaction and manipulation
+Note that in this simplified example only *one* data source is given to the SDK. In practice, multiple data sources may be passed in simultatenously. The SDK 
+handles mapping data sources to the correct plugins, and each plugin sees as input a *list* of data sources for which it advertises support. 
 
+With these high-level interfaces defined, we will now examine each component in greater detail. 
 
-# Driver
 
-The Driver's main objective is to render data for a user to consume and interact with. The user can select any data source which will be sent down to the SDK, which will in turn return tables for visualization.
-We recommend using Windows Performance Analyzer (WPA) to interact with the data as it grants a plethora of tools for analysis.
+## The SDK Driver
 
+The SDK driver's main objective is to render data for a user to observe and interact with. The driver can provide the SDK with arbitrary data sources 
+to process. For GUI drivers, these data sources are typically generated by a user asking the driver to open one or more files. However, 
```

Comment:
Ah, I see why I was confused now. I thought "Data source" is the file path like an object, and then once it is opened by the driver, it becomes "Data" but you are saying still the data source.

Reply:
Yep. I see that I can clarify this a bit by changing this last sentence to say "these data sources are typically file paths given by a user asking..." instead of saying "generated"